ipyparallel/client/asyncresult.py

```diff
@@ -134,6 +134,27 @@ def _init_futures(self):
         self._output_future.add_done_callback(self._resolve_output)
         self.add_done_callback(self._finalize_result)
 
+    @staticmethod
+    def io_stream_callback(ar, eid, msg):
+        msg_type = msg['header']['msg_type']
+        if msg_type == 'stream':
+            msg_content = msg['content']
+            stream_name = msg_content['name']
+            if stream_name == 'stdout':
+                ar._display_stream(msg_content['text'], '[stdout:%i] ' % eid)
+            elif stream_name == 'stderr':
+                ar._display_stream(
+                    msg_content['text'], '[stderr:%i] ' % eid, file=sys.stderr
+                )
+
+    def _enable_streaming_output(self):
+
+        from functools import partial
+
+        for eid, msg_id in zip(self._targets, self.msg_ids):
```

Comment:
Only change I wuld make here: make it a [context manager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) so it can be used as:

```python
with ar.stream_output():
    ...
```

where at the end of the context, the callbacks will be removed. I think it should also be a public method.

Reply:
Makes sense, done.

Reply:
Removing callbacks seems to throw a `KeyError` (on the msg_id).